inspircd: update to 3.0.1

crater2150 opened the pull request.

daniel-eys commented:
* `do_configure()` needs some adjustments to reflect the changes from v2 to v3:
```
Cannot enable m_geoip.cpp : No such file or directory in src/modules/extra
Unknown option: enable-openssl
Unknown option: enable-gnutls
Unknown option: enable-epoll
```

* `/var/lib/inspircd` needs to be created via `make_dirs` as well.

* The makefile installs `inspircd.service` which is not needed.

* `manual-dir` must be defined in `./configure` else it defaults to `/usr/manuals`.

SaberUK commented:
Can you also please pass `--distribution-label` with a value that specifically identifies the packaged version used. I assume this would be something like `--distribution-label voidlinux-${revision}`. This helps us identify the version people are using when they ask for help.

crater2150 commented:
Ok, I think I fixed all the things you mentioned.

The Travis build still fails, but I can't see, what causes the `configure` to abort. The output only contains [a warning](https://travis-ci.org/void-linux/void-packages/jobs/531371493#L497), which says to use the `--gid` switch, which I also added. Locally, the package builds and runs for me.

daniel-eys commented:
* `ERROR: Cannot open config file: /etc/inspircd/inspircd.conf` so a default config-file should be provided and defined with `conf_files="/etc/inspircd/inspircd.conf"`. (see for example how debian is doing that).
* cross-compiling is failing:
 ```
 A suitable C++ compiler could not be detected on your system!
 Set the CXX environment variable to the path to a C++ compiler binary if this is incorrect.
 ```

crater2150 commented on srcpkgs/inspircd/files/inspircd/run:
```diff
@@ -1,5 +1,4 @@
 #!/bin/sh
 install -d -m0755 -o inspircd -g inspircd /run/inspircd
```
I checked on the server where I have the current version running and it doesn't seem to use this folder. Removing it.

crater2150 commented:
I'm not sure if a default config file like Debian's, which uses a default password of "12345" for the root account, is a good idea. There are other packages, that only ship a default configuration in `/usr/share`, which the user has to copy and edit to enable it.

What do you think about adding a check for the existence of a config to the runscript, pointing to the example files if it doesn't exist?

@SaberUK is there an option in your `configure` file to disable checking the compiler by executing a compiled file? This prevents cross-compiling, as the compiler outputs a binary for a different architecture than the system building it.

SaberUK commented:
>@SaberUK is there an option in your configure file to disable checking the compiler by executing a compiled file? This prevents cross-compiling, as the compiler outputs a binary for a different architecture than the system building it.

There is not currently an option for this as we need to detect the features that the system supports.

daniel-eys commented:
> I'm not sure if a default config file like Debian's, which uses a default password of "12345" for the root account, is a good idea. There are other packages, that only ship a default configuration in `/usr/share`, which the user has to copy and edit to enable it.
> 
> What do you think about adding a check for the existence of a config to the runscript, pointing to the example files if it doesn't exist?

This sounds reasonable.
Maybe just drop in an `INSTALL.msg` instructing the user to setup an initial config file and pointing to the provided examples.
